fluss-rpc/src/main/java/com/alibaba/fluss/rpc/RpcGatewayCache.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2024 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.rpc;
+
+import com.alibaba.fluss.cluster.ServerNode;
+import com.alibaba.fluss.exception.FlussRuntimeException;
+import com.alibaba.fluss.shaded.guava32.com.google.common.cache.Cache;
```

Comment:
Suggest use caffeine cache for better cache hit rate.
For example PartialUpdaterCache